rosdiagnostic/src/rosdiagnostic/rosdiagnostic.py

```diff
@@ -0,0 +1,234 @@
+        self._print_begin_banner(msg)
+        for diag in sorted(self._get_leaf_statuses(msg.status), key=lambda d: d.level, reverse=True):
+            if prog.search(diag.name):
+                self._print_status(diag)
+        self._print_end_banner(msg)
+
+    def _print_status(self, diag):
+        if self._options.min_level <= diag.level:
+            print('[ {} ] {} - {}'.format(self._lvl_strings_map[diag.level], diag.name, diag.message))
+            if self._options.detail:
+                print('    hardware_id: {}'.format(diag.hardware_id))
+                for kv in diag.values:
+                    print('    - {}: {}'.format(kv.key, kv.value))
+
+
+def _rosdiagnostic_cmd_echo(argv):
+    parser = argparse.ArgumentParser(description='CPR ROS Diagnostics')
+    parser.add_argument('-f', '--follow', action='store_true', dest='follow',
+                        help='follows the diagnostic messages continuously')
+
+    parser.add_argument('-l', action='store', metavar='LEVEL', type=int, default=DiagnosticStatus.WARN,
+                        dest='min_level',
+                        help='the minimum diagnostic level to display (OK=0, WARN=1, ERROR=2, STALE=3) [default=1]')
+
+    parser.add_argument('--namespace', action='store', metavar='NAMESPACE', type=str, default='~', dest='ns',
+                        help='root namespace where to look for the "/diagnostics_agg" topic')
+
+    parser.add_argument('--filter', action='store', metavar='FILTER', type=str, default='.*', dest='filter',
+                        help='regular expression to be applied as a filter to the diagnostic name')
+
+    parser.add_argument('--nocolor', action='store_true',
+                        help='output should not make use of any color')
+
+    parser.add_argument('--detail', action='store_true',
+                        help='printing the full diagnostic details ')
+
+    args = parser.parse_args(argv)
+
+    _check_master()
+    rospy.init_node('rosdiagnostic', anonymous=True)
+    rosdiag = RosDiagnostics(args)
+    rospy.spin()
+    del rosdiag
+
+
+def _rosdiagnostic_cmd_version(prog):
+    stack_name = os.path.basename(__file__)
+    rosstack = rospkg.RosStack()
+    try:
+        version = rosstack.get_stack_version(stack_name)
+    except rospkg.ResourceNotFound:
+        try:
+            # hack to make it work with wet packages
+            mm = ManifestManager(PACKAGE_FILE)
+            path = mm.get_path(stack_name)
+            package_manifest = os.path.join(path, 'package.xml')
+            if os.path.exists(package_manifest):
+                from xml.etree.ElementTree import ElementTree
+                try:
+                    root = ElementTree(None, package_manifest)
+                    version = root.findtext('version')
+                except Exception:
+                    pass
+        except rospkg.ResourceNotFound:
+            version = '(unknown)'
+
+    print("{} {}".format(prog, version))
+    return version
```

Comment:
I know you've just copied this over from [rosversion](https://github.com/ros-infrastructure/rospkg/blob/master/scripts/rosversion), but it's a lot of logic for what it does. IMO not worth it— none of the other standard rosxxxxx tools do this, so the expectation of a user would be that if they want to know what version this tool is, they would do `rosversion rosdiagnostic` same as they would do `rosversion rostopic`.